pkg/plugins/common/kustomize/v1/scaffolds/internal/templates/config/prometheus/role.go

```diff
@@ -0,0 +1,68 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package prometheus
+
+import (
+	"path/filepath"
+
+	"sigs.k8s.io/kubebuilder/v3/pkg/machinery"
+)
+
+var _ machinery.Template = &Role{}
+
+// Role scaffolds a file that defines the prometheus role
+type Role struct {
+	machinery.TemplateMixin
+}
+
+// SetTemplateDefaults implements file.Template
+func (f *Role) SetTemplateDefaults() error {
+	if f.Path == "" {
+		f.Path = filepath.Join("config", "prometheus", "role.yaml")
+	}
+
+	f.TemplateBody = roleTemplate
+
+	return nil
+}
```

Comment:
How can we check the issue + that it solves the scenario in the e2e tests: See: https://github.com/kubernetes-sigs/kubebuilder/blob/master/test/e2e/v3/plugin_cluster_test.go#L255-L260
?

Reply:
I took a look at the tests and I think instead of just testing that we can access the service `/metrics` endpoint, https://github.com/prometheus-operator/kube-prometheus should be deployed instead of the Prometheus Operator, and then we should validate that the results of a request to `http://prometheus-k8s.monitoring.svc:9090/api/v1/targets` should contain the service monitor